### PR TITLE
Add SAM path setting

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Add setting to preferences

## Related Issue(s)
#222 

## Screenshots (if appropriate)
screen shot 2018-09-24 at 12 14 18 pm

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.settings
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.ServiceManager
+
+class AwsApplicationSettings : PersistentStateComponent<AwsApplicationConfiguration> {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt]: I think we probably might want to separate this into SAM settings - breaking it up down the line won't be backwards compatible and as the toolkit grows we may want to break this up.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): AwsApplicationSettings = ServiceManager.getService(
+            AwsApplicationSettings::class.java)
+    }
+}
+
+data class AwsApplicationConfiguration(
+    var samLocation: String = ""
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt]: nit: can we can something more descriptive than `location` it's the executable path right?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt]: Ya, easy enough. Will just rename this to SamSettings and also add storage annotations for aws.xml

File: jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt]: samExecutablePath?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsApplicationSettings.kt]: SamSettings now

**Pull request merged**